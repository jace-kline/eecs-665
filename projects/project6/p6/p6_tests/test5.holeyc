int f(int a, int b) {
    return a + b;
}

int main(int argc, charptr argv) {
    int a;
    int b;
    int c;
    return f(f(argc, a), f(b, c));
}