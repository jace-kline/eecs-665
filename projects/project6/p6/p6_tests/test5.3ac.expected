[BEGIN GLOBALS]
[END GLOBALS]
[BEGIN f LOCALS]
a (formal)
b (formal)
tmp0 (tmp)
[END f LOCALS]
lbl_fun_f:  enter f
            getarg 1 [a]
            getarg 2 [b]
            [tmp0] := [a] ADD64 [b]
            setret [tmp0]
            goto lbl_0
lbl_0:      leave f
[BEGIN main LOCALS]
argc (formal)
argv (formal)
a (local)
b (local)
c (local)
tmp0 (tmp)
tmp1 (tmp)
tmp2 (tmp)
[END main LOCALS]
lbl_main:   enter main
            getarg 1 [argc]
            getarg 2 [argv]
            setarg 1 [argc]
            setarg 2 [a]
            call f
            getres [tmp0]
            setarg 1 [b]
            setarg 2 [c]
            call f
            getres [tmp1]
            setarg 1 [tmp0]
            setarg 2 [tmp1]
            call f
            getres [tmp2]
            setret [tmp2]
            goto lbl_1
lbl_1:      leave main